python/kserve/kserve/model.py

```diff
@@ -75,19 +90,43 @@ def __init__(self, name: str):
         self._http_client_instance = None
         self._grpc_client_stub = None
 
-    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR):
-        request = await self.preprocess(body) if inspect.iscoroutinefunction(self.preprocess) \
-            else self.preprocess(body)
+    async def __call__(self, body, model_type: ModelType = ModelType.PREDICTOR, headers={}):
+        request_id = headers.get("X-Request-Id")
```

Comment:
@yuzisun changed the key from "X-RequestID" to "X-Request-Id" -- this needs to be tested before merging. seems like the header PR is almost ready to merge.

Reply:
done